### ISSUE TITLE
False positive `unused-private-member` when using `__new__`

### ISSUE THREAD

**Issue by larme** [PyCQA/pylint#4668]
This may relate to #4657

### Steps to reproduce
Given a file `sample2.py`:

```python
class UnusedPrivateMember:
    """unused-private-member (W0238)"""

    def __new__(cls, func, *args):
        obj = super(UnusedPrivateMember, cls).__new__(cls)
        obj.func = func
        obj.__args = args
        return obj

    def exec(self):
        return self.func(*self.__args)
```

### Current behavior

Result of `pylint sample2.py`:
```
************* Module sample2
sample2.py:7:8: W0238: Unused private member `UnusedPrivateMember.__args` (unused-private-member)
```

### Expected behavior

I would expect no unused-private-member warning

### pylint --version output

Result of `pylint --version` output:

```
pylint 3.0.0-a4
astroid 2.6.2
Python 3.7.3 (default, Jan 22 2021, 20:04:44)
[GCC 8.3.0]
```

**Comment by yushao2** [comment]
I'll look into this over the course of next week
